apps/hash-graph/bin/cli/src/subcommand/server.rs

```diff
@@ -94,207 +98,35 @@ pub struct ServerArgs {
-            )
-            .expect("failed to construct base URL"),
-            version: 1,
-        },
-        "Empty List".to_owned(),
-        Some("An Empty List".to_owned()),
-        "array".to_owned(),
-        HashMap::from([("const".to_owned(), json!([]))]),
-    );
-
-    // TODO: Revisit once an authentication and authorization setup is in place
-    let root_account_id = AccountId::new(Uuid::nil());
-
-    let mut connection = pool
-        .acquire()
-        .await
-        .change_context(GraphError)
-        .map_err(|err| {
-            tracing::error!(?err, "Failed to acquire database connection");
-            err
-        })?;
-
-    // TODO: When we improve error management we need to match on it here, this will continue
-    //  normally even if the DB is down
-    if connection
-        .insert_account_id(root_account_id)
-        .await
-        .change_context(GraphError)
-        .is_err()
-    {
-        tracing::info!(%root_account_id, "tried to create root account, but id already exists");
-    } else {
-        tracing::info!(%root_account_id, "created root account id");
-    }
 
-    // Seed the primitive data types if they don't already exist
-    for data_type in [text, number, boolean, empty_list, object, null] {
-        let title = data_type.title().to_owned();
-
-        let data_type_metadata =
-            OntologyElementMetadata::External(ExternalOntologyElementMetadata::new(
-                data_type.id().clone().into(),
-                ProvenanceMetadata::new(RecordCreatedById::new(root_account_id)),
-                OffsetDateTime::now_utc(),
-            ));
+    /// Starts a server without connecting to the type fetcher
+    #[clap(long, default_value_t = false, conflicts_with_all = ["type_fetcher_host", "type_fetcher_port"])]
+    pub offline: bool,
+}
 
-        if let Err(error) = connection
-            .create_data_type(data_type, &data_type_metadata)
-            .await
-            .change_context(GraphError)
-        {
-            if error.contains::<VersionedUrlAlreadyExists>() {
-                tracing::info!(%root_account_id, "tried to insert primitive {title} data type, but it already exists");
-            } else {
-                return Err(error.change_context(GraphError));
-            }
-        } else {
-            tracing::info!(%root_account_id, "inserted the primitive {title} data type");
+async fn restore_builtin_types<C: AsClient>(mut store: SnapshotStore<C>) -> Result<(), GraphError> {
+    let restore_result = store
+        .restore_snapshot(
+            FramedRead::new(
+                io::BufReader::new(Cursor::new(include_str!("../../built-in-types.jsonl"))),
+                codec::JsonLinesDecoder::default(),
+            ),
+            10_000,
+        )
+        .await;
+
+    if let Err(err) = restore_result {
+        if err.contains::<VersionedUrlAlreadyExists>() {
```

Comment:
What happens if the snapshot that's being used is updated. As in, let's say they change the timestamps, or update the types to point to newer versions.. I think it'll still _work_ fine, but might have some unexpected behaviour..

I don't think we want to try and work around that right now, but we might want to find a way to capture that possibility somewhere, in documentation of the option or something.

Reply:
They would be ignored as the data already exists in the database. That's the same behavior as before.

Reply:
To my understanding, this _isn't_ the same behaviour as before because we're allowing the hard-coded control over more information about the type here. We're also inserting temporal metadata and `fetchedAt` and such.

To re-emphasise a part of my point:
> As in, **let's say they change the timestamps**, or update the types to point to newer versions.. I think it'll still work fine, but might have some unexpected behaviour..

You would have a snapshot that you try and insert, there's a happy log message saying "didn't insert it because it already exists" and no warning that the data you _think_ you're inserting is actually distinct from the data that's already in the store.

Reply:
The offline mode is not designed for being used in production but _**really**_ only for testing. The only point why I kept the offline mode is because you asked for it, removing it entirely would have been a whole lot easier. Honestly, I don't like to sink any more into this if not strictly required because the whole snapshot loading here is only as a workaround if you need to wipe your database often locally and you don't have a stable internet connection.

Reply:
In the conversation about this, I suggested I don't disagree with us removing it if it adds too much burden, which you seem to be suggesting here
> removing it entirely would have been a whole lot easier.

As I've expressed before, an offline mode is _not_ only useful for development or testing, it is entirely reasonable to think someone might want to run the graph in a ring-fenced environment without access to the external web. We are not actively prioritising that use-case right now (which is why I acknowledge we may want to abandon the capability for the moment) but those use-cases have been part of the reasoning from the very first time we discussed offline support.

If we are changing the code, I'd like us to change it to code that works in the way we would want it to work, rather than something with inconsistencies that can catch us out later on. If we do not have the capacity to implement this properly at the moment, then we should remove it altogether.